src/use-force-update.ts

```diff
@@ -1,21 +1,13 @@
-import { useState } from 'react';
+import { useReducer } from 'react';
 
-interface VoidFunction {
-  (): void;
-}
+type VoidFunction = () => void
+type VoidFunctionCreator = () => VoidFunction;
+type ToggleReducer = (state: boolean, action: void) => boolean;
 
-interface VoidFunctionCreator {
-  (): VoidFunction;
-}
+const reducer: ToggleReducer = state => !state;
 
-const toggle = (state: boolean): boolean => !state;
-
-const useForceUpdate: VoidFunctionCreator = (): VoidFunction => {
-  const [ , setState ] = useState<boolean>(true);
-  const forceUpdate: VoidFunction = (): void => {
-    setState(toggle);
-  };
-  return forceUpdate;
-};
+const useForceUpdate: VoidFunctionCreator = () => (
+  useReducer(reducer, true)[1] as VoidFunction
```

Comment:
It's not as short or memory efficient, but I think destructuring first to `[ , dispatch ]` will make the repo more readable and inviting to contributors. `f(x, y)[z]` looks like a lot of magic that may not be intuitive to learners.

Why is dispatch being overridden as a `VoidFunction`? What is it's default return type?

Reply:
> It's not as short or memory efficient, but I think destructuring first to `[ , dispatch ]` will make the repo more readable and inviting to contributors. `f(x, y)[z]` looks like a lot of magic that may not be intuitive to learners.

:100:% agreed.

> Why is dispatch being overridden as a `VoidFunction`? What is it's default return type?

yeah, I saw that the assignment was a bit of a hack.
I'm fairly new to TypeScript, and don't know how to solve that. 
It doesn't compile without overriding it.

the type is `React.Dispatch<void>`.

```
Type 'Dispatch<void>' is not assignable to type 'VoidFunction'.
```

I can make the hook return this type, but that, I think, is exposing implementation details, and it's incompatible with `VoidFunction`  because it receives an argument.

Reply:
We can replace `VoidFunction` with `Dispatch<void>`.

I think
* `import { Dispatch, useReducer } from 'react';`
* delete `VoidFunctionCreator`
* change `useForceUpdate`'s definition to be `(): Dispatch<void>`